tests/xharness/Program.cs

```diff
@@ -72,6 +72,15 @@ public static int Main (string[] args)
 				{ "periodic-command-arguments=", "Arguments to the command to execute periodically.", (v) => harness.PeriodicCommandArguments = v },
 				{ "periodic-interval=", "An interval (in minutes) between every attempt to execute the periodic command.", (v) => harness.PeriodicCommandInterval = TimeSpan.FromMinutes (double.Parse (v)) },
 				{ "include-system-permission-tests:", "If tests that require system permissions (which could cause the OS to launch dialogs that hangs the test) should be executed or not. Default is to include such tests.", (v) => harness.IncludeSystemPermissionTests = ParseBool (v, "include-system-permission-tests") },
+				{ "xml-jargon:", "The xml format to be used for test results. Values can be nunitv2, nunitv3, xunit,", (v) =>
+					{
+						if (Enum.TryParse<Jargon> (v, out var jargon))
+							harness.XmlJargon = jargon;
+						else
+							harness.XmlJargon = Jargon.NUnitV3; // default
```

Comment:
I would show an error here instead, that way invalid values won't actually work.

Reply:
or at least a warning, mentioning something else (than the user-supplied value) is being used

but since it's an internal (as in not-shipped) tool I think an error is simpler - and we want to know (not learn later) that things are not like expected.